exchange_wrapper_standards: Investigate intermittent race

Intermittent race spotted here: https://github.com/thrasher-corp/gocryptotrader/actions/runs/16259232574/job/45900962328#step:11:500

Stack trace:
<details>

```console
==================
WARNING: DATA RACE
Read at 0x00c00061bd70 by goroutine 3627:
  reflect.typedmemmove()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/mbarrier.go:213 +0x0
  reflect.packEface()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:136 +0xc5
  reflect.valueInterface()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:1513 +0x179
  reflect.Value.Interface()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:1484 +0x266
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:250 +0x1a7
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Previous write at 0x00c00061bd70 by goroutine 3336:
  sync/atomic.AddInt32()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/race_amd64.s:281 +0xb
  sync/atomic.AddInt32()
      <autogenerated>:1 +0x14
  sync.(*Mutex).Unlock()
      /opt/hostedtoolcache/go/1.24.4/x64/src/sync/mutex.go:65 +0x28
  context.(*cancelCtx).propagateCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:494 +0x23e
  context.WithDeadlineCause()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:634 +0x194
  context.WithDeadline()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:617 +0x7d9
  net/http.setRequestCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:366 +0x7ec
  net/http.send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:257 +0x89c
  net/http.(*Client).send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:180 +0x14c
  net/http.(*Client).do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:728 +0x1338
  net/http.(*Client).Do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:587 +0x56
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*client).do()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/client.go:121 +0x3b
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).doRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:198 +0x42d
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).SendPayload()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:97 +0x14d
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).SendAuthenticatedHTTPRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:913 +0x31e
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelMultipleFuturesOpenOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:2315 +0x699
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelBatchOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio_wrapper.go:1148 +0x12bb
  runtime.call64()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/asm_amd64.s:777 +0x42
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:368 +0xb5
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:238 +0xc4
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3627 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3336 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44
==================
==================
WARNING: DATA RACE
Read at 0x00c0016ee300 by goroutine 3627:
  reflect.maplen()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/map_swiss.go:293 +0x0
  reflect.Value.lenNonSlice()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:1772 +0x1e4
  reflect.Value.Len()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:1761 +0x117
  internal/fmtsort.Sort()
      /opt/hostedtoolcache/go/1.24.4/x64/src/internal/fmtsort/sort.go:56 +0x104
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:816 +0x1136
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:853 +0x1d0c
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:866 +0x8a4
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:910 +0x236c
  fmt.(*pp).printArg()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:759 +0xb64
  fmt.(*pp).doPrintf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:1074 +0x5bc
  fmt.Sprintf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:239 +0x5c
  testing.(*common).Errorf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1110 +0x89
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:255 +0x51a
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Previous write at 0x00c0016ee300 by goroutine 3336:
  runtime.mapassign()
      /opt/hostedtoolcache/go/1.24.4/x64/src/internal/runtime/maps/runtime_swiss.go:191 +0x0
  context.(*cancelCtx).propagateCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:492 +0x22f
  context.WithDeadlineCause()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:634 +0x194
  context.WithDeadline()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:617 +0x7d9
  net/http.setRequestCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:366 +0x7ec
  net/http.send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:257 +0x89c
  net/http.(*Client).send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:180 +0x14c
  net/http.(*Client).do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:728 +0x1338
  net/http.(*Client).Do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:587 +0x56
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*client).do()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/client.go:121 +0x3b
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).doRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:198 +0x42d
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).SendPayload()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:97 +0x14d
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).SendAuthenticatedHTTPRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:913 +0x31e
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelMultipleFuturesOpenOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:2315 +0x699
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelBatchOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio_wrapper.go:1148 +0x12bb
  runtime.call64()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/asm_amd64.s:777 +0x42
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:368 +0xb5
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:238 +0xc4
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3627 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3336 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44
==================
==================
WARNING: DATA RACE
Write at 0x00c0016ee300 by goroutine 3333:
  runtime.mapassign()
      /opt/hostedtoolcache/go/1.24.4/x64/src/internal/runtime/maps/runtime_swiss.go:191 +0x0
  context.(*cancelCtx).propagateCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:492 +0x22f
  context.WithDeadlineCause()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:634 +0x194
  context.WithDeadline()
      /opt/hostedtoolcache/go/1.24.4/x64/src/context/context.go:617 +0x7d9
  net/http.setRequestCancel()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:366 +0x7ec
  net/http.send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:257 +0x89c
  net/http.(*Client).send()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:180 +0x14c
  net/http.(*Client).do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:728 +0x1338
  net/http.(*Client).Do()
      /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:587 +0x56
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*client).do()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/client.go:121 +0x3b
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).doRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:198 +0x42d
  github.com/thrasher-corp/gocryptotrader/exchanges/request.(*Requester).SendPayload()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/request/request.go:97 +0x14d
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).SendAuthenticatedHTTPRequest()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:913 +0x31e
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelMultipleOptionOpenOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio.go:3360 +0x652
  github.com/thrasher-corp/gocryptotrader/exchanges/gateio.(*Gateio).CancelAllOrders()
      /home/runner/work/gocryptotrader/gocryptotrader/exchanges/gateio/gateio_wrapper.go:1222 +0xce7
  runtime.call32()
      /opt/hostedtoolcache/go/1.24.4/x64/src/runtime/asm_amd64.s:776 +0x42
  reflect.Value.Call()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/value.go:368 +0xb5
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:238 +0xc4
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Previous read at 0x00c0016ee300 by goroutine 3627:
  reflect.mapIterStart()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/map_swiss.go:184 +0x0
  reflect.(*MapIter).Next()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/map_swiss.go:353 +0x112
  reflect.(*MapIter).Next()
      /opt/hostedtoolcache/go/1.24.4/x64/src/reflect/map_swiss.go:353 +0x112
  internal/fmtsort.Sort()
      /opt/hostedtoolcache/go/1.24.4/x64/src/internal/fmtsort/sort.go:59 +0x273
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:816 +0x1136
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:853 +0x1d0c
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:866 +0x8a4
  fmt.(*pp).printValue()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:910 +0x236c
  fmt.(*pp).printArg()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:759 +0xb64
  fmt.(*pp).doPrintf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:1074 +0x5bc
  fmt.Sprintf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/fmt/print.go:239 +0x5c
  testing.(*common).Errorf()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1110 +0x89
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.CallExchangeMethod()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:255 +0x51a
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests.func2()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:228 +0xfb
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3333 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44

Goroutine 3627 (finished) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x8f2
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.executeExchangeWrapperTests()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:226 +0x6c8
  github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards.TestAllExchangeWrappers.func1()
      /home/runner/work/gocryptotrader/gocryptotrader/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go:74 +0x41e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x44
==================
Bitmex Asset type SPOT:
	 Enabled pairs: [ETH-USDT]
Bitmex Asset type PERPETUALCONTRACT:
	 Enabled pairs: [ETH-USD ETH-USDT]
Bitmex Asset type FUTURES:
	 Enabled pairs: [ADA-U25 ETH-U25]
Bitmex Asset type INDEX:
	 Enabled pairs: [-BAAVE -EURUSDTPI]
--- FAIL: TestAllExchangeWrappers (4.58s)
    --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests (4.42s)
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/UpdateAccountInfo-options-ETH-TEST-DELIM-usdt-20250715-3300-p (0.16s)
            testing.go:1490: race detected during execution of test
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/UpdateOrderExecutionLimits-spot-ETH-TEST-DELIM-usdt (0.30s)
            exchange_wrapper_standards_test.go:228: GateIO Func 'UpdateOrderExecutionLimits' Error: 'pair not found for tbcusdt spot'. Inputs: [{context.Background {{} {0 0}} {0xc000d64000} map[0xc000456850:{} 0xc001098a80:{} 0xc001098cb0:{} 0xc001098f50:{} 0xc001136150:{} 0xc001136230:{} 0xc0011363f0:{} 0xc001136620:{} 0xc001137110:{} 0xc001137650:{} 0xc001137c70:{} 0xc002070070:{} 0xc002070150:{} 0xc0020705b0:{} 0xc002070850:{} 0xc002070930:{} 0xc002070b60:{} 0xc002070d90:{} 0xc002071260:{} 0xc002071490:{} 0xc0020716c0:{}] <nil> <nil>} spot].
            testing.go:1490: race detected during execution of test
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/EnsureOnePairEnabled-spot-ETH-TEST-DELIM-usdt (0.00s)
            testing.go:1490: race detected during execution of test
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/CancelBatchOrders-options-ETH-TEST-DELIM-usdt-20250715-3300-p (0.14s)
            testing.go:1490: race detected during execution of test
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/CancelBatchOrders-coinmarginedfutures-BTC-TEST-DELIM-usd (0.14s)
            testing.go:1490: race detected during execution of test
        --- FAIL: TestAllExchangeWrappers/gateio_wrapper_tests/CancelAllOrders-options-ETH-TEST-DELIM-usdt-20250715-3300-p (0.16s)
            testing.go:1490: race detected during execution of test
    --- FAIL: TestAllExchangeWrappers/bitmex_wrapper_tests (6.66s)
        testing.go:1490: race detected during execution of test
FAIL
coverage: [no statements]
FAIL	github.com/thrasher-corp/gocryptotrader/cmd/exchange_wrapper_standards	274.272s
```

</details>

--- a/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go
+++ b/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go
@@ -178,6 +178,8 @@ var validWrapperParams = []reflect.Type{
 	latestRateRequest,
 }
 
+type testCtxKey string
+
 func executeExchangeWrapperTests(ctx context.Context, t *testing.T, exch exchange.IBotExchange, assetParams []assetPair) {
 	t.Helper()
 	iExchange := reflect.TypeOf(&exch).Elem()
@@ -195,7 +197,7 @@ func executeExchangeWrapperTests(ctx context.Context, t *testing.T, exch exchang
 			if slices.ContainsFunc(validWrapperParams, func(t reflect.Type) bool {
 				return input.AssignableTo(t)
 			}) {
-				assetLen = len(assetParams) - 1
+				assetLen = len(assetParams)
 				break
 			}
 		}
@@ -207,24 +209,25 @@ func executeExchangeWrapperTests(ctx context.Context, t *testing.T, exch exchang
 			e = time.Now()
 			s = e.Add(-time.Minute * 3)
 		}
-		for y := 0; y <= assetLen; y++ {
-			inputs := make([]reflect.Value, method.Type().NumIn())
-			argGenerator := &MethodArgumentGenerator{
-				Exchange:    exch,
-				AssetParams: assetParams[y],
-				MethodName:  methodName,
-				Start:       s,
-				End:         e,
-			}
-			for z := range method.Type().NumIn() {
-				argGenerator.MethodInputType = method.Type().In(z)
-				generatedArg := generateMethodArg(ctx, t, argGenerator)
-				inputs[z] = *generatedArg
-			}
-			assetY := assetParams[y].Asset.String()
-			pairY := assetParams[y].Pair.String()
-			t.Run(methodName+"-"+assetY+"-"+pairY, func(t *testing.T) {
+		for y := range assetLen {
+			ap := assetParams[y]
+			t.Run(methodName+"-"+ap.Asset.String()+"-"+ap.Pair.String(), func(t *testing.T) {
 				t.Parallel()
+				// Create a new context for each test run to avoid race conditions
+				ctx := context.WithValue(ctx, testCtxKey("test"), t.Name()) //nolint:govet // Intentional shadow
+				inputs := make([]reflect.Value, method.Type().NumIn())
+				argGenerator := &MethodArgumentGenerator{
+					Exchange:    exch,
+					AssetParams: ap,
+					MethodName:  methodName,
+					Start:       s,
+					End:         e,
+				}
+				for z := range method.Type().NumIn() {
+					argGenerator.MethodInputType = method.Type().In(z)
+					generatedArg := generateMethodArg(ctx, t, argGenerator)
+					inputs[z] = *generatedArg
+				}
 				CallExchangeMethod(t, method, inputs, methodName, exch)
 			})
 		}
@@ -245,12 +248,16 @@ func CallExchangeMethod(t *testing.T, methodToCall reflect.Value, methodValues [
 		if isUnacceptableError(t, err) != nil {
 			literalInputs := make([]any, len(methodValues))
 			for j := range methodValues {
-				if methodValues[j].Kind() == reflect.Ptr {
+				switch {
+				case methodValues[j].Type().Implements(contextParam):
+					// Errorf will use reflection on ctx and cause a race, so we need to replace it
+					literalInputs[j] = "<context>"
+				case methodValues[j].Kind() == reflect.Ptr:
 					// dereference pointers just to add a bit more clarity
 					literalInputs[j] = methodValues[j].Elem().Interface()
-					continue
+				default:
+					literalInputs[j] = methodValues[j].Interface()
 				}
-				literalInputs[j] = methodValues[j].Interface()
 			}
 			t.Errorf("%v Func '%v' Error: '%v'. Inputs: %v.", exch.GetName(), methodName, err, literalInputs)
 		}
@@ -340,8 +347,7 @@ func generateMethodArg(ctx context.Context, t *testing.T, argGenerator *MethodAr
 			OneTimePassword: "test",
 		})
 	case argGenerator.MethodInputType.Implements(contextParam):
-		// Need to deploy a context.Context value as nil value is not
-		// checked throughout codebase.
+		// Need to deploy a context.Context value as nil value is not checked throughout codebase
 		input = reflect.ValueOf(ctx)
 	case argGenerator.MethodInputType.AssignableTo(feeBuilderParam):
 		input = reflect.ValueOf(&exchange.FeeBuilder{

